ethers-core/src/types/transaction/eip712.rs

```diff
@@ -0,0 +1,276 @@
+            Token::Uint(U256::from(keccak256(&self.name))),
+            Token::Uint(U256::from(keccak256(&self.version))),
+            Token::Uint(self.chain_id),
+            Token::Address(self.verifying_contract),
+        ])))
+    }
+}
+
+// Parse the AST of the struct to determine the domain attributes
+impl From<&syn::DeriveInput> for Eip712Domain {
+    fn from(input: &syn::DeriveInput) -> Eip712Domain {
+        let mut domain = Eip712Domain::default();
+
+        let attributes = input.attrs.first().expect("missing macro arguments");
+
+        let is_segment_valid = attributes
+            .path
+            .segments
+            .first()
+            .map(|s| s.ident == "eip712")
+            .expect("missing eip712 macro arguments");
+
+        if !is_segment_valid {
+            panic!("invalid path segment, identity does not match 'eip712'")
+        }
+
+        let mut token_stream = attributes.tokens.clone().into_iter();
+
+        if let Some(quote::__private::TokenTree::Group(g)) = token_stream.next() {
+            let group_stream = g.stream().into_iter();
+            let mut current_arg = String::new();
+            for item in group_stream {
+                if let quote::__private::TokenTree::Ident(ident) = item {
+                    current_arg = ident.to_string();
+                } else if let quote::__private::TokenTree::Literal(literal) = item {
+                    match current_arg.as_ref() {
+                        "name" => {
+                            domain.name = literal.to_string().replace("\"", "");
+                        }
+                        "version" => {
+                            domain.version = literal.to_string().replace("\"", "");
+                        }
+                        "chain_id" => {
+                            domain.chain_id = literal
+                                .to_string()
+                                .parse::<U256>()
+                                .expect("failed to parse chain id from macro arguments");
+                        }
+                        "verifying_contract" => {
+                            domain.verifying_contract = literal
+                                .to_string()
+                                .replace("\"", "")
+                                .parse::<H160>()
+                                .expect("failed to parse verifying contract");
+                        }
+                        _ => {
+                            panic!("expected arguments: 'name', 'version', 'chain_id' and 'verifying_contract'; found: {}", current_arg);
+                        }
+                    }
+                }
+            }
+        };
+
+        domain
+    }
+}
+
+// Convert rust types to enc types. This is used in determining the type hash;
+// NOTE: this is not an exhaustive list, and there may already be an existing mapping
+// in another library.
+pub fn parse_field_type(field_type: String) -> String {
```

Comment:
use ethabi::token::Token or ParamType instead of String

https://docs.rs/ethabi/15.0.0/ethabi/token/enum.Token.html

Reply:
Thanks @mattsse! Will look into using this!

Reply:
Thanks again for this! Ended up copying this method. Deprecated the old method.